### ISSUE TITLE
Since 0.5.7: Modules/images with no base image no longer allowed, breaks inclusion to multiple base images

### ISSUE THREAD

**Issue by accwebs** [avirshup/DockerMake#28]
On DockerMake 0.5.6, modules/image definitions with no base image defined were allowed and could be included into other (complete) images.  0.5.7 appears to have changed that behavior, outputting the error "FATAL ERROR: No base image found in <imagename> dependencies"

Looking at the DockerMake code now, I think maybe allowing image definitions without a base image was accidental, but I found it really helpful for use in code re-usability.  Allow me to explain via an example:

Suppose I want to create six docker images, all that are 'variants' of each other:
- base centos images (centos 6 and 7)
- base images with Java installed (centos 6 and 7)
- base images with JDK & maven installed (centos 6 and 7)

I could start out by defining the base images:

```
"base_centos:6":
  FROM: centos:6
  build_directory: .
  build: |
    RUN ...
    RUN ...
    COPY ...
    ...

"base_centos:7":
  FROM: centos:7
  build_directory: .
  build: |
    RUN ...
    RUN ...
    COPY ...
    ...
```

Then, I could create re-usable 'module' images with no base.  Examples below:

For Oracle JRE:

```
"java_oracle:jre_8_svr":
  build_directory: .
  build: |
    RUN mkdir -p /opt/java
    COPY cache/java-1.8.0.141-1.tgz /opt/java/java-1.8.0.141-1.tgz
    RUN cd /opt/java && tar xvf java-1.8.0.141-1.tgz && rm -f java-1.8.0.141-1.tgz
    RUN chown -R root:root /opt/java/java-1.8.0.141-1
    ...
```

and for Oracle JDK:

```
"java_oracle:jdk_8":
  build_directory: .
  build: |
    RUN mkdir -p /opt/java
    COPY cache/jdk-1.8.0.141-1.tgz /opt/java/jdk-1.8.0.141-1.tgz
    ...
```

and for maven:

```
"maven:3.3.9":
  build_directory: .
  build: |
    RUN mkdir -p /opt/maven
    COPY cache/apache-maven-3.3.9-bin.tar.gz /opt/maven/apache-maven-3.3.9-bin.tar.gz
    ...
```

Finally, I can now mix-and-match on assembling the base images together with optional 'modules' to create a set of variants:

```
"base_centos:6_java_8":
  requires:
    - base_centos:6
    - java_oracle:jre_8_svr
  build_directory: .

"base_centos:7_java_8":
  requires:
    - base_centos:7
    - java_oracle:jre_8_svr
  build_directory: .

"dev_centos:6_java_8":
  requires:
    - base_centos:6
    - java_oracle:jdk_8
    - maven:3.3.9
  build_directory: .
  build:
    RUN yum install -y git subversion

"dev_centos:7_java_8":
  requires:
    - base_centos:7
    - java_oracle:jdk_8
    - maven:3.3.9
  build_directory: .
  build:
    RUN yum install -y git subversion
```

See how I've basically re-used the same installation instructions for Java and Maven (without having to copy-paste)?  This no longer works in 0.5.7.

Is there some way we can tweak the code to get this type of functionality back, or better yet, officially support this type of inclusion using a different syntax?  What do you think?

**Comment by avirshup** [comment]
@accwebs - Yikes! That's a bug, that behavior should not have changed! I actually do the same thing in my own builds ...

I think I know what broke, I'll take a look and add a test for it ...

**Comment by accwebs** [comment]
Awesome!  Thanks!

**Comment by avirshup** [comment]
And thanks for finding this - it solved the issue of my own failing builds. Just released 0.5.8, which hopefully fixes it. Please LMK if not!
